packaging/linux/common/khiops-env

```diff
@@ -65,13 +65,13 @@ then
     fi
 fi
 
-# Java configuration
-# JAVA_HOME automatic setting
-
-# Set java environment for GUI if Khiops GUI is installed
-if [[ -f "/usr/share/khiops/khiops.jar" && -n "$JAVA_HOME" ]]; then
-   KHIOPS_CLASSPATH=/usr/share/khiops/norm.jar:/usr/share/khiops/khiops.jar
-   KHIOPS_JAVA_PATH=$(find -L "$JAVA_HOME" -name libjvm.so | xargs dirname)
+# Set java environment if Khiops GUI is installed
+if [ -f "/usr/share/khiops/khiops.jar" ]; then
+    if [ -z "$JAVA_HOME" ]; then 
+        JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:bin/java::" )
```

Comment:
As this computation would be done dynamically at each `khiops` launch (as `khiops` sources `khiops-env`), then an issue might occur IMHO, if Java is not installed for some reason, or otherwise removed by the user at some time after installing khiops.

Indeed, in this case `/usr/bin/java` would not exist, hence this line would set `JAVA_HOME` to `/usr/`. Hence, this incurs the risk of failing later on in this script, when trying to set `KHIOPS_JAVA_PATH`, as this would launch `find` inside `/usr/` and either fail with "Permission denied" if the user is not root, or with "dirname: missing operand" otherwise (as `libjvm.so` would not be found).

Is this intended?

Reply:
what do you think about this:
```bash
# Set java environment if Khiops GUI is installed
if [ -f "/usr/share/khiops/khiops.jar" ]; then
    if [ -z "$JAVA_HOME" ] && [ -f /usr/bin/java ]; then
        JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:bin/java::" )
    fi
    if [ -n "$JAVA_HOME" ]; then
        KHIOPS_JAVA_PATH=$(find -L "$JAVA_HOME" -name libjvm.so | xargs dirname)
        KHIOPS_CLASSPATH=/usr/share/khiops/norm.jar:/usr/share/khiops/khiops.jar
    else
        echo "Java not found. Khiops cannot launch a graphical interface. Please install a Java Runtime Environment (>= 1.8)" 
    fi
fi
```
I think it handles the problem

Reply:
IMHO, there is still one situation that I think we need to account for and also send a (specific?) warning message:

if the user sets `JAVA_HOME` to some bogus value before launching Khiops, e.g. the user sets `JAVA_HOME` to `/foo/bar/baz` which does not exist, then the second `if` would execute directly, bringing us back to the previous issue (bogus `JAVA_HOME`).

Perhaps at least checking that the directory pointed to by `JAVA_HOME` exists? And if not, warn the user specifically that `JAVA_HOME` is bogus? Or perhaps use the same warning message that Java cannot be found (but at least do not fail later on?)

Reply:
Yes indeed, with this new test I think it would be OK
```bash
# Set java environment if Khiops GUI is installed
if [ -f "/usr/share/khiops/khiops.jar" ]; then
    if [ -z "$JAVA_HOME" ] && [ -f /usr/bin/java ]; then 
        JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:bin/java::" )
    fi
    if [ ! -d "$JAVA_HOME" ]; then
        echo "the JAVA_HOME directory doesn't exist (${JAVA_HOME})"
        JAVA_HOME=""
    fi
    if [ -n "$JAVA_HOME" ]; then
        KHIOPS_JAVA_PATH=$(find -L "$JAVA_HOME" -name libjvm.so | xargs dirname)
        KHIOPS_CLASSPATH=/usr/share/khiops/norm.jar:/usr/share/khiops/khiops.jar
    else
        echo "Java not found. Khiops cannot launch a graphical interface. Please install a Java Runtime Environment (>= 1.8)" 
    fi
fi
```

Reply:
Yes, this seems to cover all cases IMHO. Closing the discussion thread.

I would just start the non-existing `JAVA_HOME` directory warning message with a capital letter, and end it with a full stop. I would also end the second warning message with a full stop.

Reply:
Also, I think `unset JAVA_HOME` also works, instead of assigning it to the empty string (because it's unset locally to the script).

Reply:
All your comments are in the new commit !